src/server/routes/suppliers.js

```diff
@@ -26,22 +29,30 @@ var createSuppliers = function(req, res)
     if(exists) {
       return res.status('409').json({ message : 'A supplier already exists' });
     } else {
-      Supplier.create(newSupplier).then(supplier => {
-        const userId = supplier.createdBy;
-        const supplierId = supplier.supplierId;
-        req.ocbesbn.serviceClient.put('user', '/users/' + userId + '?tokenUpdate=true', { supplierId: supplierId, status: 'registered' })
-          .spread(() => res.status('200').json(supplier))
-          .catch(error => {
-            switch (error.response.statusCode) {
-              case 404:
-                Supplier.delete(supplierId).then(() => null);
-                res.status(404).json({ message : 'User not found!' });
-                break;
-              default:
-                res.status(400).json({ message : error.message });
-            };
-          });
-      });
+      newSupplier.status = 'new';
+      Supplier.create(newSupplier)
+        .then(supplier => this.events.emit(supplier, 'supplier').then(() => supplier))
+        .then(supplier => {
+          const userId = supplier.createdBy;
+          const supplierId = supplier.supplierId;
+          req.ocbesbn.serviceClient.put('user', '/users/' + userId + '?tokenUpdate=true', { supplierId: supplierId, status: 'registered' })
+            .spread(() => {
```

Comment:
Because of this https://github.com/OpusCapitaBusinessNetwork/service-client#ocbesbn-service-client